### ISSUE TITLE
Example of using resource files to provide route descriptions?

### ISSUE THREAD

**Issue by dylanbeattie** [NancyFx/Nancy#2329]
Hey,

I'm looking at using the default IRouteDescriptionProvider to dynamically create SwaggerUI metadata for a Nancy API. I understand from the docs that the default implementation will look in resource files for route descriptions. In https://github.com/NancyFx/Nancy/blob/master/src/Nancy/Routing/DefaultRouteDescriptionProvider.cs

it says

    /// <summary>
    /// Default implementation of the <see cref="IRouteDescriptionProvider"/> interface. Will look for
    /// route descriptions in resource files. The resource files should have the same name as the module
    /// for which it defines routes.
    /// </summary>

If I add MyModule.resx to my solution, VS2013 automatically creates a MyModule.Designer.cs codebehind file that causes a naming conflict with the module. I can fix this by setting the "Custom Tool" to blank on the `.resx` file, which removes the resource code-behind file - but then when I enumerate across the route collection exposed by IRouteCacheProvider.GetCache(), the Description property on all my RouteDescriptions is the empty string.

Is there an example somewhere of using the default RouteDescriptionProvider to expose API documentation like this?

Thanks!

**Comment by jchannon** [comment]
Have you seen this?
https://www.nuget.org/packages/Nancy.Swagger/0.1.0-alpha3

You can specify meta data in a module see here
https://github.com/khellang/Nancy.Swagger/tree/master/samples/Nancy.Swagger.Demo/Modules

Not exactly what you asked but might give you another option

**Comment by dylanbeattie** [comment]
Yeah, I've had a look at that, but wasn't keen on the requirement to maintain a separate set of metadata for each module. I'm more interested in how much of that data can be determined directly from the module's own routing registration, so there's less onus on developers to update metadata when making changes to a module signature. I'm currently hacking around on something like this:

    var cache = cacheProvider.GetCache();
    var routes = cache.SelectMany(pair => pair.Value.Select(tuple => tuple.Item2));
    dynamic docs = new ExpandoObject();
    docs.swagger = "2.0";
    docs.info = new {
        version = "1.0",
        title = "Spotlight Agency API"
    };
    docs.paths = (dynamic)(new ExpandoObject {
    });
    var paths = new ExpandoObject() as IDictionary<string, object>;
    foreach (var path in routes.GroupBy(r => r.Path)) {
        var methods = new ExpandoObject() as IDictionary<string, object>;
        foreach (var method in path) {
            var args = method.Segments.Where(segment => Regex.IsMatch(segment, @"\{.*\}"));
            var argDocs = args.Select(arg => new {
                name = arg.Split(':')[0],
                @in = "path",
                @type = "string" // TODO: parse type constraint
            });
            methods.Add(method.Method.ToLowerInvariant(), new {
                description = method.Description,
                parameters = argDocs
            });
        }
        paths[path.Key] = methods;
    }

    docs.paths = paths;
    return (docs);

but as I said, it's returning blank route descriptions and I'm not entirely sure why.

**Comment by khellang** [comment]
> I've had a look at that, but wasn't keen on the requirement to maintain a separate set of metadata for each module.

Yeah, that's a large part of the problem. Because of Nancy's dynamic nature, it's Real Hard™ to get good metadata out, without putting that burden on the dev. That's partly why I lost interest in trying to get a decent Swagger implementation out for Nancy...

**Comment by danoftheman** [comment]
@dylanbeattie Did you ever get anywhere with this?
